servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,184 @@
+
+  public enum AuthorizationRuleType {
+    CREATE_REFERENCE,
+    DELETE_REFERENCE,
+    LIST_COMMIT_LOG,
+    READ_ENTRIES,
+    ASSIGN_REFERENCE_TO_HASH,
+    COMMIT_CHANGE_AGAINST_REFERENCE,
+    READ_ENTITY_VALUE,
+    UPDATE_ENTITY,
+    DELETE_ENTITY;
+  }
+
+  @Inject
+  public CelAccessChecker(
+      QuarkusNessieAuthorizationConfig config, CompiledAuthorizationRules compiledRules) {
+    this.config = config;
+    this.compiledRules = compiledRules;
+  }
+
+  @Override
+  public void canCreateReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.CREATE_REFERENCE);
+  }
+
+  @Override
+  public void canAssignRefToHash(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.ASSIGN_REFERENCE_TO_HASH);
+  }
+
+  @Override
+  public void canDeleteReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.DELETE_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntries(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.READ_ENTRIES);
+  }
+
+  @Override
+  public void canListCommitLog(AccessContext context, NamedRef ref) throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.LIST_COMMIT_LOG);
+  }
+
+  @Override
+  public void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws AccessControlException {
+    canPerformOpOnReference(context, ref, AuthorizationRuleType.COMMIT_CHANGE_AGAINST_REFERENCE);
+  }
+
+  @Override
+  public void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.READ_ENTITY_VALUE);
+  }
+
+  @Override
+  public void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.UPDATE_ENTITY);
+  }
+
+  @Override
+  public void canDeleteEntity(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException {
+    canPerformOpOnPath(context, key, AuthorizationRuleType.DELETE_ENTITY);
+  }
+
+  private void canPerformOpOnReference(
+      AccessContext context, NamedRef ref, AuthorizationRuleType type) {
+    if (!config.enabled()) {
+      return;
+    }
+    boolean allowed =
+        compiledRules.getRules().entrySet().stream()
+            .anyMatch(
+                entry -> {
+                  try {
+                    return entry
+                        .getValue()
+                        .execute(
+                            Boolean.class,
+                            ImmutableMap.of(
```

Comment:
How can we check role or direct group memberships? Mean, updating all security policies when users come and go isn't great. The default implementation in this class may return an empty set, but implementations of this class for concrete servers can easily do extract role/group memberships, which makes security-policy-checks easier.

Reply:
I agree that this isn't great but for this implementation I don't think we were planning to do group membership checks and such